src/writeToFileSystem.ts

```diff
@@ -0,0 +1,50 @@
+import { promises as fs } from "fs";
+import {
+  SerializedPageCodeReport,
+  RepoCodeReport,
+  serializeRepoCodeReport,
+} from "./CodeExampleTypes";
+
+// Get the code blocks on a given page
+export const writeCodeToFile = async (
+  value: string,
+  baseDir: string,
+  filePath: string
+) => {
+  try {
+    await fs.mkdir(baseDir, { recursive: true });
+    await fs.writeFile(filePath, value);
+  } catch (error) {
+    process.exit(1);
+  }
+};
+
+export const writeRepoReportToFile = async (repoCodeReport: RepoCodeReport) => {
+  const outputBaseDir = `output/code-example-reports/`;
+  const outputPath = `output/code-example-reports/report-${repoCodeReport.repo}.json`;
+  const serializedReport = serializeRepoCodeReport(repoCodeReport);
+  try {
+    const jsonString = JSON.stringify(serializedReport, null, 2);
+    await fs.mkdir(outputBaseDir, { recursive: true });
+    await fs.writeFile(outputPath, jsonString, "utf-8");
+  } catch (error) {
+    console.log("Error writing report to file: %s", error);
```

Comment:
Note that log goes to stdout while error goes to stderr, which can make a different when piping commands to each other or redirecting to file (e.g. `somecommand > log.txt` <-- only stdout goes to log.txt, stderr still written to console)
```suggestion
    console.error("Error writing report to file: %s", error);
```

Reply:
Ooo, good point, thanks!